### ISSUE TITLE
Category mismatch during product import (CSV) when the category has a different slug than its name.

### ISSUE THREAD

**Issue by danilosimonatto** [polylang/polylang#1560]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [x] This bug happens with only Polylang plugin active
- [x] This bug happens with a default WordPress theme active
- [x] I can reproduce this bug consistently

## Environment

- Polylang Pro 3.6.4
- Polylang Polylang for WooCommerce 2.1
- The language is set from the directory name in pretty permalinks
- Share slugs Activated

## Steps to reproduce the Issue

1. Create a WooCommerce category.
2. Append "-something" to the slug of the category you want to import the products into, e.g., "cat-something."
3. Import a CSV product file with the Categories column filled with the category name, e.g., "Cat."

## Expected behavior and actual behavior

Products should be placed inside the specified category, regardless of the "-text" appended to the slug.

On my production website, I had many category slugs with text appended, like "shirts-woman" and "shirts-man." When I imported a CSV product file, it generated duplicate categories multiple times, e.g., "shirts-woman-1, shirts-woman-2, shirts-woman-3," each with one product. That's also a bit strange because, during testing, it generated only one category without the appended slug text, e.g., "shirts," with all the imported products inside.

In short, it seems that the category is not matched during product import if the category slug isn't the same as the category name.

This took me a lot of testing and caused a headache before I realized I could have just changed the slugs.

**Comment by Hug0-Drelon** [comment]
Hi @danilosimonatto,

Thanks for your feedback. That said, any support request for our paid plugins must be made through our helpdesk  [here](https://polylang.pro/documentation/support/contact-support/).
